perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow.py

```diff
@@ -351,12 +357,94 @@ def GetAvgCpuUtilization(self, start_time: datetime, end_time: datetime):
     # Multiply fractional cpu util by 100 to display a percentage usage
     return round(self._GetAvgValueFromTimeSeries(results) * 100, 2)
 
+  def GetMaxOutputThroughput(self, ptransform: str,
+      start_time: datetime, end_time: datetime):
+    """Get max throughput from a particular pTransform during job run interval.
+
+    Args:
+      start_time: datetime specifying the beginning of the time interval.
+      end_time: datetime specifying the end of the time interval.
+
+    Returns:
+      Max value across time interval
+    """
+    client = monitoring_v3.MetricServiceClient()
+    project_name = f'projects/{self.project}'
+
+    now_seconds = int(time.time())
+    end_time_seconds = int(end_time.timestamp())
+    # Dataflow metrics data can take up to 180 seconds to appear
+    if (now_seconds - end_time_seconds) < DATAFLOW_METRICS_DELAY_SECONDS:
+      logging.info(
+          'Waiting for Dataflow metrics to be available (up to 3 minutes)...')
+      time.sleep(
+          DATAFLOW_METRICS_DELAY_SECONDS - (now_seconds - end_time_seconds))
+
+    interval = TimeInterval()
+    # Shift TZ of datetime arguments since FromDatetime() assumes UTC
+    # See
+    # https://googleapis.dev/python/protobuf/latest/google/protobuf/timestamp_pb2.html#google.protobuf.timestamp_pb2.Timestamp.FromDatetime
+    interval.start_time.FromDatetime(
+        start_time.astimezone(datetime.timezone.utc))
+    interval.end_time.FromDatetime(
+        end_time.astimezone(datetime.timezone.utc))
+
+    api_filter = (
+        'metric.type = "dataflow.googleapis.com/job/elements_produced_count" '
+        f'AND resource.labels.project_id = "{self.project}" '
+        f'AND metric.labels.job_id = "{self.job_id}" '
+        f'AND metric.labels.ptransform = "{ptransform}" ')
+
+    aggregation = Aggregation(
+        alignment_period={'seconds': 60},  # 1 minute
+        per_series_aligner=Aggregation.Aligner.ALIGN_RATE,
+    )
+
+    results = client.list_time_series(
+        name=project_name,
+        filter_=api_filter,
+        interval=interval,
+        view=enums.ListTimeSeriesRequest.TimeSeriesView.FULL,
+        aggregation=aggregation,
+    )
+
+    if not results:
+      logging.warning(
+          'No monitoring data found. Unable to calculate max throughput.')
+      return None
+
+    return self._GetMaxValueFromTimeSeries(results)
+
+  def GetSubscriptionBacklogSize(self, subscription_name, interval_length=4):
```

Comment:
Move this method to the child class. It doesn't belong here conceptually.

Reply:
Similar to GetJobStatus, this was placed in parent class since it applies here as well e.g. in case of a streaming job (with pubsub input) that is executed from jar file.

Reply:
I think we shouldn't polute this class with PubSub stuff. Let's better leave that role for the subclass.

Reply:
Resolving as per our conversation